huaweicloud/resource_huaweicloud_bms_instance_test.go

```diff
@@ -0,0 +1,144 @@
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"github.com/huaweicloud/golangsdk/openstack/bms/v1/baremetalservers"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+)
+
+func TestAccBmsInstance_basic(t *testing.T) {
+	var instance baremetalservers.CloudServer
+
+	rName := fmt.Sprintf("tf-acc-test-%s", acctest.RandString(5))
+	resourceName := "huaweicloud_bms_instance.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheckBms(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckBmsInstanceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccBmsInstance_basic(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckBmsInstanceExists(resourceName, &instance),
+					resource.TestCheckResourceAttr(resourceName, "name", rName),
+					resource.TestCheckResourceAttr(resourceName, "status", "ACTIVE"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckBmsInstanceDestroy(s *terraform.State) error {
+	config := testAccProvider.Meta().(*config.Config)
+	bmsClient, err := config.BmsV1Client(HW_REGION_NAME)
+	if err != nil {
+		return fmt.Errorf("Error creating HuaweiCloud bms client: %s", err)
+	}
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "huaweicloud_bms_instance" {
+			continue
+		}
+
+		server, err := baremetalservers.Get(bmsClient, rs.Primary.ID).Extract()
+		if err == nil {
+			if server.Status != "SOFT_DELETED" {
+				return fmt.Errorf("Instance still exists")
+			}
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckBmsInstanceExists(n string, instance *baremetalservers.CloudServer) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No ID is set")
+		}
+
+		config := testAccProvider.Meta().(*config.Config)
+		bmsClient, err := config.BmsV1Client(HW_REGION_NAME)
+		if err != nil {
+			return fmt.Errorf("Error creating HuaweiCloud bms client: %s", err)
+		}
+
+		found, err := baremetalservers.Get(bmsClient, rs.Primary.ID).Extract()
+		if err != nil {
+			return err
+		}
+
+		if found.ID != rs.Primary.ID {
+			return fmt.Errorf("Instance not found")
+		}
+
+		*instance = *found
+
+		return nil
+	}
+}
+
+func testAccBmsInstance_basic(rName string) string {
+	return fmt.Sprintf(`
+
+resource "huaweicloud_compute_keypair" "test" {
+  name = "%s"
+  public_key = "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDAjpC1hwiOCCmKEWxJ4qzTTsJbKzndLo1BCz5PcwtUnflmU+gHJtWMZKpuEGVi29h0A/+ydKek1O18k10Ff+4tyFjiHDQAT9+OfgWf7+b1yK+qDip3X1C0UPMbwHlTfSGWLGZquwhvEFx9k3h/M+VtMvwR1lJ9LUyTAImnNjWG7TAIPmui30HvM2UiFEmqkr4ijq45MyX2+fLIePLRIFuu1p4whjHAQYufqyno3BS48icQb4p6iVEZPo4AE2o9oIyQvj2mx4dk5Y8CgSETOZTYDOR3rU2fZTRDRgPJDH9FWvQjF5tA0p3d9CoWWd2s6GKKbfoUIi8R/Db1BSPJwkqB jrp-hp-pc"
```

Comment:
public_key include hard code and it's used to import the keypair, can we use lifecycle to ignore the state check?

Reply:
fixed